### PR TITLE
[pg] Allow QueryConfig based queries rather than just string queries

### PR DESCRIPTION
**Pull request by jgwmaxwell**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/brianc/node-postgres/wiki/Client#method-query-parameterized
- [x] Increase the version number in the header if appropriate.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dslint/dt.json" }`.

### CODE REVIEW

**Comment by jgwmaxwell** [comment]
The current set of options or querying doesn't allow for `parameterized` queries - this adds that in.

File: types/pg/index.d.ts [hunk 1]
```diff
@@ -73,7 +73,7 @@ export declare class Pool extends events.EventEmitter {
     end(): Promise<void>;
 
     query(queryText: string, values?: any[]): Promise<QueryResult>;
-
+    query(queryText: QueryConfig, callback: (err: Error, result: QueryResult) => void): void;
     query(queryText: string, callback: (err: Error, result: QueryResult) => void): void;
```
Comment (jgwmaxwell) [inline types/pg/index.d.ts]: Thanks, I'll address this. Looking at the spec for client queries it's missing another couple of bits too so I'll fix it properly.

**Comment by mhegazy** [comment]
@pspeter3 can you please review

**Comment by pspeter3** [comment]
lgtm

**Pull request merged**